Recent opdescs declaration change causes compilation failures on aarch64

# Summary
The changes in c071b9f6ea9cc39385f73b783e634648ad69af4b mean that oneDNN cannot be compiled with gcc or clang.

# Version
commit c071b9f6ea9cc39385f73b783e634648ad69af4b

# Environment
oneDNN includes hardware-specific optimizations and may behave
differently on depending on the compiler and build environment. Include
the following information to help reproduce the issue:
* CPU make and model: Arm (R) Neoverse (R) V1
* OS version : 21~22.04.1-Ubuntu SMP Thu Nov  7 17:35:43 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux
* Compiler version: gcc-10, gcc-12, clang-18
* CMake version: 3.22.1
* git hash: c071b9f6ea9cc39385f73b783e634648ad69af4b

# Steps to reproduce
```
cmake -S ./ -B "./build" \
        -DCMAKE_BUILD_TYPE=Debug \
        -DCMAKE_CXX_COMPILER_LAUNCHER="ccache" \
        -DCMAKE_C_COMPILER_LAUNCHER="ccache" \
        -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
        -DDNNL_AARCH64_USE_ACL=1 \
        -DDNNL_BUILD_FOR_CI=ON \
        -DDNNL_ENABLE_JIT_PROFILING=0 \
        -DDNNL_OMP_RUNTIME=0 \
        -DDNNL_CPU_RUNTIME="OMP" \
        -DDNNL_TEST_SET=NIGHTLY \
        -DONEDNN_BUILD_GRAPH=0 \

cd ./build
make -j $(nproc)
```

You can use [verbose mode](https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html)
and [benchdnn](https://github.com/oneapi-src/oneDNN/tree/master/tests/benchdnn)
to validate correctness of all primitives the library supports. If this does not
work a short C/C++ program or modified unit tests demonstrating the issue
will greatly help with the investigation.

# Observed behavior
```
pu::aarch64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu::aarch64::sve_512, dnnl_f32>::pd_t; Args = {const dnnl::impl::op_desc_t*&, const dnnl_primitive_attr*&, const dnnl::impl::convolutio
n_fwd_pd_t*&}]’:                                                                                                                                                                                   
/home/sidmen01/oneDNN/src/common/primitive_desc.hpp:463:9:   required from ‘static std::unique_ptr<_Tp> dnnl::impl::primitive_desc_t::make_unique_pd(Args&& ...) [with pd_t = dnnl::impl::cpu::aarc
h64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu::aarch64::sve_512, dnnl_f32>::pd_t; Args = {const dnnl::impl::op_desc_t*&, const dnnl_primitive_attr*&, const dnnl::impl::convolution_fwd_pd
_t*&}]’                                                                                                                                                                                            
/home/sidmen01/oneDNN/src/common/primitive_desc.hpp:476:40:   required from ‘static dnnl::impl::status_t dnnl::impl::primitive_desc_t::create(dnnl::impl::primitive_desc_t**, const dnnl::impl::op_
desc_t*, const dnnl::impl::primitive_attr_t*, dnnl::impl::engine_t*, const dnnl::impl::primitive_desc_t*) [with pd_t = dnnl::impl::cpu::aarch64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu:
:aarch64::sve_512, dnnl_f32>::pd_t; dnnl::impl::status_t = dnnl_status_t; dnnl::impl::primitive_attr_t = dnnl_primitive_attr; dnnl::impl::engine_t = dnnl_engine]’                                 
/home/sidmen01/oneDNN/src/common/impl_list_item.hpp:93:27:   required from ‘constexpr dnnl::impl::impl_list_item_t::impl_list_item_t(type_deduction_helper_t<pd_t>) [with pd_t = dnnl::impl::cpu::a
arch64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu::aarch64::sve_512, dnnl_f32>::pd_t]’                                                                                                     
/home/sidmen01/oneDNN/src/cpu/cpu_convolution_list.cpp:120:86:   required from here                                                                                                                
/home/sidmen01/oneDNN/src/common/primitive_desc.hpp:462:75: error: invalid conversion from ‘const dnnl::impl::op_desc_t*’ to ‘const dnnl::impl::convolution_desc_t*’ [-fpermissive]                
  462 |             pd_t_compat(Args &&...args) : pd_t(std::forward<Args>(args)...) {}                                                                                                             
      |                                                                           ^                                                                                                                
      |                                                                           |                                                                                                                
      |                                                                           const dnnl::impl::op_desc_t*                                                                                     
In file included from /home/sidmen01/oneDNN/src/cpu/aarch64/jit_sve_1x1_convolution.hpp:35,                                                                                                        
                 from /home/sidmen01/oneDNN/src/cpu/cpu_convolution_list.cpp:63:                                                                                                                   
/home/sidmen01/oneDNN/src/cpu/aarch64/jit_uni_dw_convolution.hpp:110:40: note:   initializing argument 1 of ‘dnnl::impl::cpu::aarch64::jit_uni_dw_convolution_bwd_data_t<isa, diff_dst_type, diff_s
rc_type>::pd_t::pd_t(const dnnl::impl::convolution_desc_t*, const dnnl::impl::primitive_attr_t*, const dnnl::impl::convolution_fwd_pd_t*) [with dnnl::impl::cpu::aarch64::cpu_isa_t isa = dnnl::imp
l::cpu::aarch64::sve_512; dnnl_data_type_t diff_dst_type = dnnl_f32; dnnl_data_type_t diff_src_type = dnnl_f32; dnnl::impl::primitive_attr_t = dnnl_primitive_attr]’                               
  110 |         pd_t(const convolution_desc_t *adesc, const primitive_attr_t *attr,                                                                                                                
      |              ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~                                                                                                                                               
/home/sidmen01/oneDNN/src/common/primitive_desc.hpp: In instantiation of ‘dnnl::impl::primitive_desc_t::make_unique_pd(Args&& ...)::pd_t_compat::pd_t_compat(Args&& ...) [with pd_t = dnnl::impl::c
pu::aarch64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu::aarch64::sve_256, dnnl_f32>::pd_t; Args = {const dnnl::impl::op_desc_t*&, const dnnl_primitive_attr*&, const dnnl::impl::convolutio
n_fwd_pd_t*&}]’:                                                                                                                                                                                   
/home/sidmen01/oneDNN/src/common/primitive_desc.hpp:463:9:   required from ‘static std::unique_ptr<_Tp> dnnl::impl::primitive_desc_t::make_unique_pd(Args&& ...) [with pd_t = dnnl::impl::cpu::aarc
h64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu::aarch64::sve_256, dnnl_f32>::pd_t; Args = {const dnnl::impl::op_desc_t*&, const dnnl_primitive_attr*&, const dnnl::impl::convolution_fwd_pd
_t*&}]’                                                                                                                                                                                            
/home/sidmen01/oneDNN/src/common/primitive_desc.hpp:476:40:   required from ‘static dnnl::impl::status_t dnnl::impl::primitive_desc_t::create(dnnl::impl::primitive_desc_t**, const dnnl::impl::op_
desc_t*, const dnnl::impl::primitive_attr_t*, dnnl::impl::engine_t*, const dnnl::impl::primitive_desc_t*) [with pd_t = dnnl::impl::cpu::aarch64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu:
:aarch64::sve_256, dnnl_f32>::pd_t; dnnl::impl::status_t = dnnl_status_t; dnnl::impl::primitive_attr_t = dnnl_primitive_attr; dnnl::impl::engine_t = dnnl_engine]’                                 
/home/sidmen01/oneDNN/src/common/impl_list_item.hpp:93:27:   required from ‘constexpr dnnl::impl::impl_list_item_t::impl_list_item_t(type_deduction_helper_t<pd_t>) [with pd_t = dnnl::impl::cpu::a
arch64::jit_uni_dw_convolution_bwd_data_t<dnnl::impl::cpu::aarch64::sve_256, dnnl_f32>::pd_t]’                                                                                                     
/home/sidmen01/oneDNN/src/cpu/cpu_convolution_list.cpp:120:86:   required from here                          
```

# Expected behavior
I would very much like it to compile

--- a/src/cpu/aarch64/jit_uni_dw_convolution.hpp
+++ b/src/cpu/aarch64/jit_uni_dw_convolution.hpp
@@ -107,9 +107,7 @@ template <cpu_isa_t isa, data_type_t diff_dst_type,
         data_type_t diff_src_type = diff_dst_type>
 struct jit_uni_dw_convolution_bwd_data_t : public primitive_t {
     struct pd_t : public cpu_convolution_bwd_data_pd_t {
-        pd_t(const convolution_desc_t *adesc, const primitive_attr_t *attr,
-                const convolution_fwd_pd_t *hint_fwd_pd)
-            : cpu_convolution_bwd_data_pd_t(adesc, attr, hint_fwd_pd), jcp_() {}
+        using cpu_convolution_bwd_data_pd_t::cpu_convolution_bwd_data_pd_t;
 
         DECLARE_COMMON_PD_T(JIT_IMPL_NAME_HELPER("jit_dw:", jcp_.isa, ""),
                 jit_uni_dw_convolution_bwd_data_t);
@@ -193,10 +191,9 @@ template <cpu_isa_t isa, data_type_t src_type,
         data_type_t diff_weights_type = src_type>
 struct jit_uni_dw_convolution_bwd_weights_t : public primitive_t {
     struct pd_t : public cpu_convolution_bwd_weights_pd_t {
-        pd_t(const convolution_desc_t *adesc, const primitive_attr_t *attr,
-                const convolution_fwd_pd_t *hint_fwd_pd)
-            : cpu_convolution_bwd_weights_pd_t(adesc, attr, hint_fwd_pd)
-            , jcp_() {}
+        using cpu_convolution_bwd_weights_pd_t::
+                cpu_convolution_bwd_weights_pd_t;
+
         using jit_uni_dw_convolution_bwd_weights
                 = jit_uni_dw_convolution_bwd_weights_t<isa, src_type,
                         diff_weights_type>;

